File: lib/models/everything_response.dart
import 'package:news_app/models/article.dart';

class EverythingResponse {
  final String status;
  final int totalResults;
  final List<Article> articles;

  EverythingResponse({
    required this.status,
    required this.totalResults,
    required this.articles,
  });

  factory EverythingResponse.fromJson(Map<String, dynamic> json) =>
      EverythingResponse(
        status: json['status'],
        totalResults: json['totalResults'],
        articles: (json['articles'] as List<dynamic>)
            .map((e) => Article.fromJson(e as Map<String, dynamic>))
            .toList(),
      );
}

File: android/app/src/main/java/it/fudeo/news_app/models/EverythingResponse.java
package it.fudeo.news_app.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import it.fudeo.news_app.Pigeon;

public class EverythingResponse {

    @SerializedName("status")
    private String status;

    @SerializedName("totalResults")
    private int totalResults;

    @SerializedName("articles")
    private List<Pigeon.Article> articles;

    public EverythingResponse(String status, int totalResults, List<Pigeon.Article> articles) {
        this.status = status;
        this.totalResults = totalResults;
        this.articles = articles;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public int getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(int totalResults) {
        this.totalResults = totalResults;
    }

    public List<Pigeon.Article> getArticles() {
        return articles;
    }

    public void setArticles(List<Pigeon.Article> articles) {
        this.articles = articles;
    }
}
